mycpp/leaky_mylib.h

```diff
@@ -125,12 +125,60 @@ class Writer : public Obj {
   virtual bool isatty() = 0;
 };
 
+class Buf {
+ public:
+  Buf(char* data, int len) : data_(data), len_(len) {}
+  void reset() {
+      free(data_);
+      data_ = nullptr;
+      len_ = 0;
+  }
+  operator bool() const {
+    return data_ != nullptr;
+  }
+  void expand(Str* s) {
+    return expand(s->data_, ::len(s));
+  }
+  void expand(const char* s, int n) {
+    int orig_len = len_;
+    len_ += n;
+
+    data_ = static_cast<char*>(realloc(data_, len_ + 1));
```

Comment:
Looking at this more closely, how does this `data_` buffer get freed in the garbage collected case?

Remember there were 2 cases:

1. the global case with `gBuf`, used by format strings, where you can call reset()
2. the garbage collected mylib.BufWriter instances used by `asdl/format.py`.  The garbage collector can't call `reset()`

I think it was `_bin/cxx-asan/mycpp/examples/parse.mycpp` that exposed this, or maybe it was in containers

Anyway we can talk about it tonight, thanks again!